scripts/verticalmenu.js

```diff
@@ -0,0 +1,227 @@
+  },
+  {
+    name: "lighter",
+    imgurl: "./assets/blinkit1.jpg",
+    price: "40",
+    discountedPrice: "30",
+    id: "smokes",
+    quantity: "1 piece",
+  },
+  {
+    name: "tamatos",
+    imgurl: "./assets/blinkit5.jpg",
+    price: "100",
+    discountedPrice: "70",
+    id: "vegies",
+    quantity: "1 kg",
+  },
+  {
+    name: "onion",
+    imgurl: "./assets/blinkit2.jpg",
+    price: "140",
+    discountedPrice: "130",
+    id: "vegies",
+    quantity: "1 kg",
+  },
+  {
+    name: "Butter oil",
+    imgurl: "./assets/blinkit3.jpg",
+    price: "1000",
+    discountedPrice: "700",
+    id: "milk products",
+    quantity: "300 ml",
+  },
+  {
+    name: "Green kwei",
+    imgurl: "./assets/blinkit6.jpg",
+    price: "170",
+    discountedPrice: "120",
+    id: "snacks",
+    quantity: "4 pieces",
+  },
+  {
+    name: "Fortune Aata",
+    imgurl: "./assets/blinkit7.jpg",
+    price: "250",
+    discountedPrice: "200",
+    id: "snacks",
+    quantity: "1kg",
+  },
+  {
+    name: "lays",
+    imgurl: "./assets/blinkit8.jpg",
+    price: "40",
+    discountedPrice: "20",
+    id: "snacks",
+    quantity: "1 pack",
+  },
+  {
+    name: "limka",
+    imgurl: "./assets/blinkit9.jpg",
+    price: "60",
+    discountedPrice: "50",
+    id: "drinks",
+    quantity: "750 ml",
+  },
+];
+
+function genereateHTMLElement(elementType) {
+  return document.createElement(elementType);
+}
+
+function filterarray(filterid = "all") {
+  console.log(filterid);
+  if (filterid === "all") return allProducts;
+  const filteredarray = [];
+  allProducts.map((product) => {
+    if (product.id === filterid) {
+      filteredarray.push(product);
+    }
+  });
+  console.log(filteredarray);
+  return filteredarray;
+}
+
+document.addEventListener("DOMContentLoaded", function () {
+  var productList = allProducts;
+
+  document.getElementById("all").addEventListener("click", () => {
+    productList = filterarray("all");
+    generateItem(productList);
+  });
+
+  document.getElementById("vegies").addEventListener("click", () => {
+    productList = filterarray("vegies");
+    generateItem(productList);
+  });
+
+  document.getElementById("smokes").addEventListener("click", () => {
+    productList = filterarray("smokes");
+    generateItem(productList);
+  });
+
+  document.getElementById("snacks").addEventListener("click", () => {
+    productList = filterarray("snacks");
+    generateItem(productList);
+  });
+
+  document.getElementById("drinks").addEventListener("click", () => {
+    productList = filterarray("drinks");
+    generateItem(productList);
+  });
+
+  document.getElementById("milk products").addEventListener("click", () => {
+    productList = filterarray("milk products");
+    generateItem(productList);
+  });
+});
+
+const generateItem = (productList) => {
+  const grid = document.getElementById("grid-container");
+
+  while (grid.hasChildNodes()) {
+    grid.removeChild(grid.firstChild);
+  }
+  productList.map((product) => {
```

Comment:
avoid inline functions.
1. Readability is difficult.
2. Unit testing is not possible